app/dbdctl.c

```diff
@@ -318,6 +320,86 @@ static int handle_reconfigure(int argc, char **argv){
 	return 0;
 }
 
+static void print_info(struct dattobd_info *infoitem)
+{
+    int i;
+    printf("minor: %d\n", infoitem->minor);
+    printf("state: %lu\n", infoitem->state);
+    printf("error: %d\n", infoitem->error);
+    printf("cache_size: %lu\n", infoitem->cache_size);
+    printf("falloc_size: %llu\n", infoitem->falloc_size);
+    printf("seqid: %llu\n", infoitem->seqid);
+    printf("uuid: ");
+    for(i = 0; i < COW_UUID_SIZE; i++)
+        printf("%02x", (unsigned int)(infoitem->uuid[i] & 0xff));
+    printf("\n");
+
+    printf("cow: %s\n", infoitem->cow);
+    printf("bdev: %s\n", infoitem->bdev);
+    printf("\n");
+}
+
+static int handle_active_devices(void){
+    int max = 255;
```

Comment:
I still think it's better to expose this information and use it. Most people aren't going to change the default value of 24 devices, so this would allocate 10x the space it needs.